dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java

```diff
@@ -1135,4 +1155,121 @@ protected boolean doesPageContentsHaveContainer(
             return false;
         }
     }
+
+    @CloseDBIfOpened
+    @Override
+    public int getContentletReferenceCount(final String contentletId) throws DotDataException {
```

Comment:
if you want to do it faster I would refactor such as 
```
 @Override
    public int getContentletReferenceCount(final String contentletId) throws DotDataException {
 
        final Optional<Integer> referenceCount = this.multiTreeCache.get().getContentletReferenceCount(contentletId);
        if (referenceCount.isPresent()) {
            return referenceCount.get();
        }
        
        this.multiTreeCache.get().putContentletReferenceCount(contentletId, count.intValue());
        return count.intValue();
    }
    
  
@CloseDBIfOpened
    private int getContentletReferenceCountDb (final String contentletId) {
    
    final Long count =
               return  ((Long) new DotConnect().setSQL(SELECT_CONTENTLET_REFERENCES).addParam(contentletId).loadObjectResults().get(0).get("count"));
                }
```

Reply:
Done!